core/src/main/kotlin/org/eln2/sim/electrical/mna/component/Resistor.kt

```diff
@@ -1,51 +1,15 @@
 package org.eln2.sim.electrical.mna.component
 
-import org.eln2.sim.electrical.mna.Circuit
+class Resistor(r: Double) : Port() {
+    override val name: String = "r"
 
-/**
- * Implements a simple, static resistor.
- *
- * The most important field is arguably [resistance]; updating this value will result in [Circuit.matrixChanged].
- */
-open class Resistor : Port() {
-    override var name: String = "r"
-    override val imageName = "resistor"
+    override var resistance: Double = r
 
-    /**
-     * The resistance of this resistor, in Ohms.
-     *
-     * Setting this will cause [Circuit.factorMatrix] to be called on the next step or substep.
-     */
-    open var resistance: Double = 1.0
-        set(value) {
-            // Remove our contribution to the matrix (using a negative resistance... should work)
-            field = -field
-            stamp()
-
-            // Add our new contribution
-            field = value
-            stamp()
-        }
```

Comment:
Just a note that removing this effectively makes resistor values unmutable. The stamp is never called again in your code, and the constructor being the only place to set the resistance is a no-no.